groupmne/tests/test_inverse.py

```diff
@@ -1,33 +1,217 @@
+    fwds = prepare_fwds(fwds, src_ref)
+
+    # Test ground metric with a wrong shape
+    M_wrong_shape = np.ones((2, 3))
+    ot_dict = dict(M=M_wrong_shape, tol=100.)
+    with pytest.raises(ValueError, match="M must be an array"):
+        compute_group_inverse(fwds, evokeds, noise_covs, method=method,
+                              spatiotemporal=False, **ot_dict)
+
+    # Test negative ground metric
+    n_features = fwds[0]["sol_group"]["data"].shape[1]
+    M_negative = - np.ones((n_features, n_features))
+    ot_dict = dict(M=M_negative, tol=100.)
+    with pytest.raises(ValueError, match="M must be non-negative"):
+        compute_group_inverse(fwds, evokeds, noise_covs, method=method,
+                              spatiotemporal=False, **ot_dict)
+
+
+@pytest.mark.parametrize("method", ["lasso", "relasso", "multitasklasso",
+                                    "dirty", "mtw", "remtw"])
+def test_evokeds(src_fwds, method):
+    src_ref, fwds = src_fwds
+    cov = mne.read_cov(cov_fname, verbose=False)
+    noise_covs = [cov, cov]
+    ev = mne.read_evokeds(ave_fname, verbose=False)[0]
+    ev0 = ev.copy().crop(tmin=0.1, tmax=0.13)
+    ev1 = ev.copy().crop(tmin=0.1, tmax=0.15)
+    ev3 = ev.copy().crop(tmin=0.12, tmax=0.15)
+    fwds = prepare_fwds(fwds, src_ref)
+
+    evokeds = [ev0, ev0, ev0]
+    with pytest.raises(ValueError, match="The number of evokeds is not equal"):
+        compute_group_inverse(fwds, evokeds, noise_covs, method=method)
+    evokeds = [ev0, ev1]
+    with pytest.raises(ValueError,
+                       match="times array with a different length"):
+        compute_group_inverse(fwds, evokeds, noise_covs, method=method)
+
+    evokeds = [ev0, ev3]
+    with pytest.raises(ValueError, match="different time coordinates"):
+        compute_group_inverse(fwds, evokeds, noise_covs, method=method)
+
+
+# def test_vs_mixed_norm(src_fwds):
+#     src_ref, fwds = src_fwds
+#     cov = mne.read_cov(cov_fname, verbose=False)
+#     noise_covs = [cov]
+#     fwds_prepared = prepare_fwds(fwds[:1], src_ref)
+#     ev = mne.read_evokeds(ave_fname, verbose=False)[0]
+#     ev.crop(tmin=0.1, tmax=0.12)
+#
+#     stc_gl = compute_group_inverse(fwds_prepared, [ev], noise_covs,
+#                                    method="multitasklasso",
+#                                    spatiotemporal=True,
+#                                    alpha=0.5)[0]
+#     stc_mx = mixed_norm(ev, fwds_prepared[0], cov, alpha=50, loose=0.,
+#                         debias=False)
+#     stc_mx.data *= 1e9
+#     stc_mx.subject = "fsaverage"
+#
+#     support_gl = np.where(stc_gl.data.any(1))[0]
+#     support_mx = np.where(stc_mx.data.any(1))[0]
+#     ampl_gl = stc_gl.data[support_gl]
+#     ampl_mx = stc_mx.data[support_mx]
+#     # vertices_gl = stc_gl.vertices[0][support_gl]
+#     # vertices_mx = stc_mx.vertices[0][support_mx]
+#
+#     # assert_array_equal(vertices_gl, vertices_mx)
+#     assert abs(ampl_gl - ampl_mx).max() < 1e-4
```

Comment:
Why commenting the test ?

Reply:
For some reason It works with the example HF data but not with the testing data, I get slightly different sources. The preprocessing of the cov in groupmne is based the one in  the sovlers of `mne.minimum_norm` but I noticed it is not the same in the sparse solvers`mne.inverse_sparse`. I need to do some digging offline before attempting this test